Fix issues when compiling with clang (where cpp is provided by gcc
in the standalone toolchain).

diff -u -r ../ruby-2.3.1/configure.in ./configure.in
--- ../ruby-2.3.1/configure.in	2016-04-24 13:44:40.000000000 -0400
+++ ./configure.in	2016-05-18 01:10:09.523385169 -0400
@@ -799,7 +799,7 @@
 rb_cv_warnflags="$warnflags"
 if test "$GCC:${warnflags+set}:no" = yes::no; then
     if test $gcc_major -ge 4; then
-	extra_warning=-Werror=extra-tokens
+	extra_warning=
     else
 	extra_warning=
     fi
@@ -813,9 +813,7 @@
 		 -Werror=pointer-arith \
 		 -Werror=write-strings \
 		 -Werror=declaration-after-statement \
-		 -Werror=shorten-64-to-32 \
 		 -Werror=implicit-function-declaration \
-		 -Werror=division-by-zero \
 		 -Werror=deprecated-declarations \
 		 -Wno-packed-bitfield-compat \
 		 $extra_warning \
